selfie.c

```diff
@@ -6582,21 +6585,17 @@ uint64_t open_write_only(char* name) {
   // we try opening write-only files using platform-specific flags
   // to make selfie platform-independent, this may nevertheless
   // not always work and require intervention
-  uint64_t fd;
-
-  if (WINDOWS)
-    // use Windows flags
-    fd = sign_extend(open(name, WINDOWS_O_BINARY_CREAT_TRUNC_WRONLY, S_IRUSR_IWUSR_IRGRP_IROTH), SYSCALL_BITWIDTH);
-  else {
-    // try Mac flags first as default
-    fd = sign_extend(open(name, MAC_O_CREAT_TRUNC_WRONLY, S_IRUSR_IWUSR_IRGRP_IROTH), SYSCALL_BITWIDTH);
+  uint64_t flags;
 
-    if (signed_less_than(fd, 0))
-      // then try Linux flags
-      fd = sign_extend(open(name, LINUX_O_CREAT_TRUNC_WRONLY, S_IRUSR_IWUSR_IRGRP_IROTH), SYSCALL_BITWIDTH);
-  }
+  if (HOST_OS == WINDOWS)
```

Comment:
We should set the correct flags in `init_system` now and then just use those.

Reply:
Yeah sure